src/Symfony/Component/Console/Attribute/Option.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Console\Attribute;
+
+use Symfony\Component\Console\Completion\CompletionInput;
+use Symfony\Component\Console\Completion\Suggestion;
+use Symfony\Component\Console\Exception\LogicException;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+
+#[\Attribute(\Attribute::TARGET_PARAMETER)]
+class Option
+{
+    private ?int $mode = null;
+    private string $typeName = '';
+
+    /**
+     * Represents a console command --option definition.
+     *
+     * If unset, the `name` and `default` values will be inferred from the parameter definition.
+     *
+     * @param array|string|null                                              $shortcut        The shortcuts, can be null, a string of shortcuts delimited by | or an array of shortcuts
+     * @param scalar|array|null                                              $default         The default value (must be null for self::VALUE_NONE)
+     * @param array|callable-string(CompletionInput):list<string|Suggestion> $suggestedValues The values used for input completion
+     */
+    public function __construct(
+        public string $name = '',
+        public array|string|null $shortcut = null,
+        public string $description = '',
+        public string|bool|int|float|array|null $default = null,
+        public array|string $suggestedValues = [],
+    ) {
+        if (\is_string($suggestedValues) && !\is_callable($suggestedValues)) {
+            throw new \TypeError(\sprintf('Argument 5 passed to "%s()" must be either an array or a callable-string.', __METHOD__));
+        }
+    }
+
+    /**
+     * @internal
+     */
+    public static function tryFrom(\ReflectionParameter $parameter): ?self
+    {
+        /** @var self $self */
+        if (null === $self = ($parameter->getAttributes(self::class, \ReflectionAttribute::IS_INSTANCEOF)[0] ?? null)?->newInstance()) {
```

Comment:
can't the `$self` be declared a line before? I know the Sf codebase doesn't shy from variable declarations in an if, but I can't help to think it's unnecessarily making it harder to read for no benefit in this case

Reply:
No strong opinion on that matter, I personally prefer this way to reduce a single goal in one line. Let see what other think about it

Reply:
All in all I'm fine with this as-is. While it's indeed hard to read, this is boilerplate that can be found at many places in core and as such doesn't deserve more than being inlined. It's a matter of preference so since we tolerate assignments in if statements, it's not worth bothering to me